samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemConsumer.java

```diff
@@ -285,6 +321,27 @@ public void register(SystemStreamPartition systemStreamPartition, String offset)
     metrics.registerTopicAndPartition(toTopicAndPartition(tp));
   }
 
+  @Override
+  public void register(SystemStreamPartition systemStreamPartition, Startpoint startpoint) {
+    if (started.get()) {
+      String exceptionMessage = String.format("KafkaSystemConsumer: %s had started. Registration of ssp: %s, startpoint: %s failed.", this, systemStreamPartition, startpoint);
+      throw new SamzaException(exceptionMessage);
+    }
+
+    if (!Objects.equals(systemStreamPartition.getSystem(), systemName)) {
+      LOG.warn("{}: ignoring SSP {}, because this consumer's system doesn't match.", this, systemStreamPartition);
+      return;
+    }
+
+    LOG.debug("Registering the ssp: {}, startpoint: {} with the consumer.", systemStreamPartition, startpoint);
+
+    super.register(systemStreamPartition, startpoint);
+
+    TopicPartition topicPartition = toTopicPartition(systemStreamPartition);
+    topicPartitionToStartpointMap.put(topicPartition, startpoint);
```

Comment:
There is logic in the old `register` which checks that the requested offset is older than the existing offset, in order to avoid missing messages. Just double checking that we want to intentionally leave that out?